Assets/Scripts/R&D Tree/ResearchNode.cs

```diff
@@ -0,0 +1,102 @@
+using UnityEngine;
+using UnityEngine.Events;
+
+public class ResearchNode : MonoBehaviour
+{
+    [SerializeField] private RDTreeManager _treeManager;
+    public float ResearchValue { get; private set; }
+
+    public bool IsResearchInQueue { get; private set; }
+
+    public ResearchNodeSetting NodeSetting { get; private set; }
+
+    public ResearchStates CurrentResearchState { get; private set; }
+
+    [HideInInspector] public UnityEvent ResearchDoneEvent;
```

Comment:
Unsure if the Node should contain the event... The node is more difficult to access since you give no option to retrieve the node in the treemanager at the moment. Expanding on the previous comment, I think it would be easier to work with indexes instead of passing the whole node class. With that, you can create a callback function like:
`TreeManager.SetFinishedCallback(1, () => { });` Where 1 is the index of the node in the list and the second argument describes the event.

Again, leave it for now with the deadline creeping up, but know this isn't the best approach :)